init-service/src/main/java/uk/ac/ox/ndph/mts/init_service/loader/Loader.java

```diff
@@ -18,30 +20,56 @@
     private final RoleServiceInvoker roleServiceInvoker;
     private final SiteServiceInvoker siteServiceInvoker;
     private final Trial trialConfig;
+    private final InitProgressReporter initProgressReporter;
 
     @Value("${delay-start:1}")
     private long delayStartInSeconds;
 
     @Autowired
     public Loader(Trial trialConfig, PractitionerServiceInvoker practitionerServiceInvoker,
-                  RoleServiceInvoker roleServiceInvoker, SiteServiceInvoker siteServiceInvoker) {
+                  RoleServiceInvoker roleServiceInvoker, SiteServiceInvoker siteServiceInvoker,
+                  InitProgressReporter initProgressReporter) {
         this.trialConfig = trialConfig;
         this.practitionerServiceInvoker = practitionerServiceInvoker;
         this.roleServiceInvoker = roleServiceInvoker;
         this.siteServiceInvoker = siteServiceInvoker;
+        this.initProgressReporter = initProgressReporter;
     }
 
     @Override
-    public void run(String... args) throws InterruptedException {
-        // TODO: remove this temporary fix in this story: https://ndph-arts.atlassian.net/browse/ARTS-362
+    public void run(String... args) throws InterruptedException, IOException {
         // Give the other services some time to come online.
+        initProgressReporter.submitProgress(String.format("init service delaying for %d seconds.",
+                delayStartInSeconds));
         Thread.sleep(delayStartInSeconds * 1000);
 
-        roleServiceInvoker.execute(trialConfig.getRoles());
-        List<String> siteIds = siteServiceInvoker.execute(trialConfig.getSites());
-        // This yuk but it is the assumption in story https://ndph-arts.atlassian.net/browse/ARTS-164
-        String siteIdForUserRoles = siteIds.get(0);
-        practitionerServiceInvoker.execute(trialConfig.getPersons(), siteIdForUserRoles);
+        try {
```

Comment:
1. I think all progress reports should also be written to the log
2. Can the actual string be moved to an enum like we have on the other services?

Reply:
1. done
2. done